### ISSUE TITLE
bigquery: Empty arrays inside records in BigQuery are returned as nil

### ISSUE THREAD

**Issue by mikus** [googleapis/google-cloud-go#9383]
#### Environment details

  - Programming language: go
  - OS: macOS 14.3
  - Language runtime version: 1.21.6
  - Package version: cloud.google.com/go/bigquery v1.57.1

#### Steps to reproduce

Assuming the below schema
```json
[
  {
    "name": "id",
    "mode": "",
    "type": "STRING"
  },
  {
    "name": "urls",
    "mode": "",
    "type": "RECORD",
    "fields": [
      {
        "name": "twitter",
        "mode": "REPEATED",
        "type": "STRING"
      },
      {
        "name": "facebook",
        "mode": "REPEATED",
        "type": "STRING"
      },
      {
        "name": "website",
        "mode": "REPEATED",
        "type": "STRING"
      }
    ]
  }
]
```

when you read records using an iterator

```golang
client, err := bigquery.NewClient(ctx, gcpProject)
it := client.Dataset(dataset).Table(table).Read(ctx)

var row map[string]bigquery.Value
err := it.Next(&row)

js, err := json.Marshal(row)
fmt.Print(string(js))
```

for empty arrays you will get `null` instead of `[]` because they are `nil` instead of `[]string{}` 

```json
{"id":"myid","urls":{"facebook":null,"twitter":null,"website":["https://mysite.com/"]}}
```

**Comment by bilus** [comment]
Making the parent RECORD required doesn't fix the problem:

```json
[
  {
    "name": "id",
    "mode": "REQUIRED",
    "type": "STRING"
  },
  {
    "name": "urls",
    "mode": "REQUIRED",
    "type": "RECORD",
    "fields": [
      {
        "name": "twitter",
        "mode": "REPEATED",
        "type": "STRING"
      },
      {
        "name": "facebook",
        "mode": "REPEATED",
        "type": "STRING"
      },
      {
        "name": "website",
        "mode": "REPEATED",
        "type": "STRING"
      }
    ]
  }
]
```

Insert record:

```SQL
INSERT INTO `project.dataset.foobar` (id, urls) VALUES ("1", STRUCT<website ARRAY<STRING>, facebook ARRAY<STRING>, twitter ARRAY<STRING>> ([], [], []))
```

```go
package main

import "cloud.google.com/go/bigquery"
import "context"
import "log"
import "encoding/json"
import "fmt"

func main() {
	ctx := context.Background()
	client, err := bigquery.NewClient(ctx, "project")
	if err != nil {
		log.Fatal(err)
	}
	it := client.Dataset("dataset").Table("foobar").Read(ctx)

	var row map[string]bigquery.Value
	err = it.Next(&row)
	if err != nil {
		log.Fatal(err)
	}

	js, err := json.Marshal(row)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Print(string(js))
}
```

Output:

```json
{"id":"1","urls":{"facebook":null,"twitter":null,"website":null}}
```

**Comment by alvarowolfx** [comment]
This is similar to https://github.com/googleapis/google-cloud-go/issues/4950 and this old PR https://github.com/googleapis/google-cloud-go/pull/7315 tries to solve it. This introduces a breaking change and we ended up postponing it and it got stale. Thoughts @shollyman ?

**Comment by alvarowolfx** [comment]
fixed by #7315
